content/400-reference/200-api-reference/050-prisma-client-reference.mdx

```diff
@@ -5597,40 +5597,39 @@ Utility types are helper functions and types that live on the Prisma namespace.
 
 The `validator` helps you create re-usable query parameters based on your schema models while making sure that the objects you create are valid. See also: [Using `Prisma.validator`](/concepts/components/prisma-client/advanced-type-safety/prisma-validator) <span class="concept"></span>
 
-#### Reference
+There are two ways you can use the `validator`:
+
+#### Using generated Prisma Client types
+
+Provides a type-level approach to validate data:
 
 ```ts
-export function validator<V>(): <S>(select: Exact<S, V>) => S
+Prisma.validator<GeneratedType>({ args })
+```
 
-type Exact<A, W = unknown> = W extends unknown
-  ? A extends Narrowable
-    ? Cast<A, W>
-    : Cast<
-        { [K in keyof A]: K extends keyof W ? Exact<A[K], W[K]> : never },
-        { [K in keyof W]: K extends keyof A ? Exact<A[K], W[K]> : W[K] }
-      >
-  : never
+#### Using a "selector"
 
-type Narrowable = string | number | boolean | bigint
+Provides a validator that uses an existing Prisma Client instance. This pattern allows you to select the model, operation, and query option to validate against.
 
-type Cast<A, B> = A extends B ? A : B
+You can also use an instance of Prisma Client that has been extended using a [Prisma Client extension](/concepts/components/prisma-client/client-extensions)<span class="concept"></span>.
 
-export const type: unique symbol
+```ts
+Prisma.validator(
+  PrismaClientInstance,
```

Comment:
Is this defined somewhere else? Might be easier to expand this example to include creating an extended client.

Reply:
The code snippet highlights the function's interface, somewhat like pseudo code. You can find a more concrete example in the Examples section.